### PR TITLE
Stasis grenade effects

### PR DESCRIPTION
**Pull request by 1upD**
This PR includes changes to separate the stasis grenade sprite color and particle effect from the hopwire/xen grenade. Instead of 0 message being sent to the client, 3 is sent instead to trigger a stasis grenade effect instead of a xen grenade effect.

Stasis grenades will create decals underneath where they were set off.

Some ConVar defaults have also been changed in this request.

### CODE REVIEW

File: sp/src/game/client/episodic/c_weapon_hopwire.cpp [hunk 1]
```diff
@@ -138,7 +143,7 @@ void C_HopwireExplosion::AddParticles( void )
 
 		// Base of the core effect
 #ifdef EZ2
-		sParticle = (SimpleParticle *) m_pSimpleEmitter->AddParticle( sizeof(SimpleParticle), m_pSimpleEmitter->GetPMaterial( "effects/XenGrenadeFlash" ), GetRenderOrigin() );
+		sParticle = (SimpleParticle *) m_pSimpleEmitter->AddParticle( sizeof(SimpleParticle), m_iExplosionType == 1 ? m_pSimpleEmitter->GetPMaterial("effects/strider_muzzle") : m_pSimpleEmitter->GetPMaterial( "effects/XenGrenadeFlash" ), GetRenderOrigin() );
```
Comment (1upD) [inline sp/src/game/client/episodic/c_weapon_hopwire.cpp]: This should be a new unique effect, not necessarily the original strider muzzle effect

File: sp/src/game/server/episodic/grenade_hopwire.cpp [hunk 2]
```diff
@@ -2850,6 +2902,11 @@ CStasisVortexController *CStasisVortexController::Create( const Vector &origin,
 	// Start the vortex working
 	pVortex->StartPull( origin, radius, strength, duration );
 
+	trace_t	tr;
+	AI_TraceLine(origin + Vector(0, 0, 1), origin - Vector(0, 0, 128), MASK_SOLID_BRUSHONLY, pVortex, COLLISION_GROUP_NONE, &tr);
+
+	UTIL_DecalTrace(&tr, "Glowbie.Puddle");
```
Comment (1upD) [inline sp/src/game/server/episodic/grenade_hopwire.cpp]: This should use an original decal, not the Glowbie puddle decal

File: sp/src/game/server/episodic/grenade_hopwire.cpp [hunk 3]
```diff
@@ -2850,6 +2902,11 @@ CStasisVortexController *CStasisVortexController::Create( const Vector &origin,
 	// Start the vortex working
 	pVortex->StartPull( origin, radius, strength, duration );
 
+	trace_t	tr;
+	AI_TraceLine(origin + Vector(0, 0, 1), origin - Vector(0, 0, 128), MASK_SOLID_BRUSHONLY, pVortex, COLLISION_GROUP_NONE, &tr);
```
Comment (1upD) [inline sp/src/game/server/episodic/grenade_hopwire.cpp:2940]: Should this be a different trace function?

**Review by 1upD — COMMENTED** [review]
I performed a self review of this code and left some comments for myself to address soon before I take this PR out of draft status

**Comment by 1upD** [comment]
Need to add unique sounds too

**Comment by 1upD** [comment]
Test comment

**Review by Blixibon — APPROVED** [review]
I'm still a bit iffy on whether it's a good long-term implementation for the clientside distinction between a stasis and hopwire grenade explosion to just be a separate user msg byte *(a 3rd hopwire type can't be added without changing the whole switch statement, for example)*, but it otherwise LGTM

**Pull request merged**